### PR TITLE
Add channels example

### PR DESCRIPTION
**Pull request by twetzel59**
Added an example that shows
how to use channels in a typical
multithreaded scenario.

### CODE REVIEW

**Comment by twetzel59** [comment]
This is just my second PR for Nim, so let me know if anything isn't quite right.

The example aims to be general but useful by explaining what I think most people might often do with ``channels``.

**Comment by timotheecour** [comment]
could that be turned into a `runnableExamples:` ? that way it's guaranteed to always compile/run

**Comment by twetzel59** [comment]
@timotheecour, can ``runnableExamples`` work at module level? I wasn't quite sure how to use it l, so I searched the Nim repo and with a quick look I only saw it inside ``proc``s. 

Many other modules have module level examples in a comment. Do they not use ``runnableExamples`` because they haven't been updates, or for a good reason?

**Comment by Araq** [comment]
This example mixes channels with ``threadpool`` and that's unwise. Also how the documentation of channels is generated is bad and needs to be changed.

**Comment by twetzel59** [comment]
Would it be better to use pure threads, or single threading instead?

**Comment by Araq** [comment]
Better show how to use plain threads with these channels.

**Comment by timotheecour** [comment]
@twetzel59 
> can runnableExamples work at module level? 

just filed: 
* runnableExamples doesn't work at module level #8641

IMO it'd be higher priority to fix this, so that you can add your example in top-level `runnableExamples` instead of adding more legacy code that'll later need to be changed to `runnableExamples` block.
But doesn't have to be you :-) If no-one volunteers I can look into it, but I have my hands full with a ton of other PR's at the moment. 
I don't think it's a hard bug to fix

File: lib/system/channels.nim [hunk 1]
```diff
@@ -15,6 +15,98 @@
...
+## =======
+## In this example, a program is created that uses the ``threadpool`` module
+## to launch two separate procedures. These procedures send strings through a
+## channel to the main thread, which calls ``echo`` on the strings.
+## Both blocking and non-blocking receives are demonstrated.
+## The channel is declared as a global to avoid the unsafety of
+## passing it by raw ``ptr``.
+##
+## .. code-block :: Nim
+##   # For simple and efficient multithreading.
```
Comment (Varriount) [inline lib/system/channels.nim]: Any particular reason there send to be a self-imposed 50-character line length here? Usually 80 is used.

Comment (twetzel59) [reply lib/system/channels.nim]: Probably VSCode :)

I'll need to adjust the over-length guard line.

**Comment by timotheecour** [comment]
> @timotheecour, can runnableExamples work at module level? I wasn't quite sure how to use it l, so I searched the Nim repo and with a quick look I only saw it inside procs.

it now works in devel

**Comment by Araq** [comment]
No progress, the example still uses the threadpool.

**Pull request closed without merging**